src/Compilers/CSharp/Test/Symbol/Symbols/AssemblyAndNamespaceTests.cs

```diff
@@ -676,5 +676,17 @@ namespace B;
                 // int x; // 1
                 Diagnostic(ErrorCode.ERR_NamespaceUnexpected, "x").WithLocation(3, 5));
         }
+
+        [Fact, WorkItem(54836, "https://github.com/dotnet/roslyn/issues/54836")]
+        public void AssemblyRetargetableAttributeIsRespected()
+        {
+            var code = @"
+using System.Reflection;
+[assembly: AssemblyFlags(AssemblyNameFlags.Retargetable)]";
+
+            var comp = CreateCompilation(code);
+            Assert.True(comp.Assembly.Identity.IsRetargetable);
```

Comment:
If I understand correctly, the main purpose of `AssemblyNameFlags.Retargetable` is to cause a flag to be emitted in metadata. 
FWIW, it looks like that was working correctly despite this bug in AssemblyIdentity (see test `AssemblyFlagsAttribute` which checks the emitted metadata).

Reply:
AssemblyFlags is just written out directly, it doesn't read from Identity at all when doing so: https://sourceroslyn.io/#Microsoft.CodeAnalysis/PEWriter/MetadataWriter.cs,1969.